liquidctl/driver/asetek.py

```diff
@@ -91,6 +111,56 @@
 # Control request type
 _USBXPRESS = usb.util.CTRL_OUT | usb.util.CTRL_TYPE_VENDOR | usb.util.CTRL_RECIPIENT_DEVICE
 
+class AsetekRuntimeSettings():
+
+    def __init__(self, color1, color2, color3, alert_temp, interval1, interval2, blackout,
+                 fading, blinking, enable_alert):
+        self.command = _CMD_RUNTIME
+        self.red_light_intensity = color1[0]
+        self.green_light_intensity = color1[1]
+        self.blue_light_intensity = color1[2]
+        self.fade_r = color2[0]
+        self.fade_g = color2[1]
+        self.fade_b = color2[2]
+        self.high_r = color3[0]
+        self.high_g = color3[1]
+        self.high_b = color3[2]
+        self.high_temp = alert_temp
+        self.fade_throb_int = interval1
+        self.throb_seq = interval2
+        self.is_light_enable = not blackout
+        self.is_fade_enable = fading
+        self.is_throb_enable = blinking
+        self.is_high_temp_enable = enable_alert
+        self.boot_me = 0x00
+        self.fan_type = 0x01
+
+    def __repr__(self):
+        return "AsetekRuntimeSettings({!r})".format(self.__dict__)
+
+    def __bytes__(self):
+        return bytes([
+            self.command,
+            self.red_light_intensity,
+            self.green_light_intensity,
+            self.blue_light_intensity,
+            self.fade_r,
+            self.fade_g,
+            self.fade_b,
+            self.high_r,
+            self.high_g,
+            self.high_b,
+            self.high_temp,
+            self.fade_throb_int,
+            self.throb_seq,
+            self.is_light_enable,
+            self.is_fade_enable,
+            self.is_throb_enable,
+            self.is_high_temp_enable,
+            self.boot_me,
+            self.fan_type
+        ])
```

Comment:
I think it probably makes sense to use this abstraction(or at least some sort of object based abstraction) for the other drivers. One advantage to using this sort of abstraction is you can easily then add methods to it that update variables without touching others for managing the state of the device, say if you want to just change the speed you could add a `def blink(self):` or something like that which updates the instance variables to enable blinking without having to change the others.

Reply:
Actually, other devices (not from the 690LC family) allow each aspect to be configured independently.  So I'm not sure a state abstraction like this would be well suited for them.

And from the point of view of liquidctl (or any software dealing with USB devices, because we can't really enforce that only a single piece of software is talking to a given device), stateless and independent configuration messages are better.